### ISSUE TITLE
Windows Build

### ISSUE THREAD

**Issue by fkromer** [welchbj/tt#6]
I tried to build the C-extensions (OS: Windows with MSVC v140 – VS 2015 C++-Buildtools (v14.00), Python: 3.7.6). However running `python setup.py build` fails with

```
> python setup.py build
running build
running build_py
running build_ext
building 'tt._clibs.picosat' extension
creating build\temp.win-amd64-3.7
creating build\temp.win-amd64-3.7\Release
creating build\temp.win-amd64-3.7\Release\Users
creating build\temp.win-amd64-3.7\Release\Users\Florian
creating build\temp.win-amd64-3.7\Release\Users\Florian\gitlab
creating build\temp.win-amd64-3.7\Release\Users\Florian\gitlab\tt
creating build\temp.win-amd64-3.7\Release\Users\Florian\gitlab\tt\tt
creating build\temp.win-amd64-3.7\Release\Users\Florian\gitlab\tt\tt\_clibs
creating build\temp.win-amd64-3.7\Release\Users\Florian\gitlab\tt\tt\_clibs\picosat
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DNDEBUG -DNGETRUSAGE -Dinline=__inline -IC:\Users\Florian\gitlab\tt\tt\_clibs\picosat -IC:\Python37\include -IC:\Python37\include "-IC:\Program Files (x86)
\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" /TcC:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c /Fobuild\temp.win-amd64-3
.7\Release\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.obj
picosat.c
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(860): warning C4244: "=": Konvertierung von "unsigned __int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1177): warning C4996: 'strcpy': This function or variable may be unsafe. Consider using strcpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1227): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1438): warning C4334: "<<": Das Ergebnis der 32-Bit-Verschiebung wurde implizit in 64 Bits konvertiert. (War eine 64-Bit-Verschiebung vorgesehen?)
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1557): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1558): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1562): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1610): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1670): warning C4244: "return": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1693): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1746): warning C4244: "return": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1825): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1845): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(1854): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2258): warning C4146: Einem vorzeichenlosen Typ wurde ein unärer Minus-Operator zugewiesen. Das Ergebnis ist weiterhin vorzeichenlos.
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2278): warning C4267: "=": Konvertierung von "size_t" nach "int", Datenverlust möglich
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2275): warning C4996: 'sprintf': This function or variable may be unsafe. Consider using sprintf_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
C:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt\stdio.h(1769): note: Siehe Deklaration von "sprintf"
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2303): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2344): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2357): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2368): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2370): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2398): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2405): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2594): warning C4244: "Initialisierung": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2594): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2634): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2721): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2785): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2805): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2803): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2926): warning C4244: "=": Konvertierung von "__int64" in "long", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(2941): warning C4244: "=": Konvertierung von "__int64" in "long", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3275): warning C4267: "=": Konvertierung von "size_t" nach "int", Datenverlust möglich
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3291): warning C4244: "=": Konvertierung von "double" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3293): warning C4146: Einem vorzeichenlosen Typ wurde ein unärer Minus-Operator zugewiesen. Das Ergebnis ist weiterhin vorzeichenlos.
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3297): warning C4244: "=": Konvertierung von "double" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3310): warning C4244: "=": Konvertierung von "double" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3314): warning C4244: "=": Konvertierung von "double" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3320): warning C4244: "=": Konvertierung von "double" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3273): warning C4996: 'sprintf': This function or variable may be unsafe. Consider using sprintf_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
C:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt\stdio.h(1769): note: Siehe Deklaration von "sprintf"
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3285): warning C4996: 'sprintf': This function or variable may be unsafe. Consider using sprintf_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
C:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt\stdio.h(1769): note: Siehe Deklaration von "sprintf"
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3511): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3659): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(3671): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(4726): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5124): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5173): warning C4244: "=": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5328): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5081): warning C4244: "Initialisierung": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5081): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5083): warning C4244: "Initialisierung": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5083): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5326): warning C4244: "Initialisierung": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5326): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5445): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5584): warning C4244: "=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5540): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5585): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(5661): warning C4244: "=": Konvertierung von "unsigned __int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(6574): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(6718): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(6754): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(6872): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(6956): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(7373): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(7378): warning C4244: "Initialisierung": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(7395): warning C4244: "Initialisierung": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(7664): warning C4244: "=": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(7725): warning C4244: "=": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(7765): warning C4244: "=": Konvertierung von "__int64" in "int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(8199): warning C4244: "+=": Konvertierung von "__int64" in "unsigned int", möglicher Datenverlust
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(8216): warning C4146: Einem vorzeichenlosen Typ wurde ein unärer Minus-Operator zugewiesen. Das Ergebnis ist weiterhin vorzeichenlos.
C:\Users\Florian\gitlab\tt\tt\_clibs\picosat\picosat.c(8248): warning C4146: Einem vorzeichenlosen Typ wurde ein unärer Minus-Operator zugewiesen. Das Ergebnis ist weiterhin vorzeichenlos.
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DNDEBUG -DNGETRUSAGE -Dinline=__inline -IC:\Users\Florian\gitlab\tt\tt\_clibs\picosat -IC:\Python37\include -IC:\Python37\include "-IC:\Program Files (x86)
\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" /TcC:\Users\Florian\gitlab\tt\tt\_clibs\picosatmodule.c /Fobuild\temp.win-amd64-3.7
\Release\Users\Florian\gitlab\tt\tt\_clibs\picosatmodule.obj
picosatmodule.c
c:\python37\include\pyconfig.h(203): fatal error C1083: Datei (Include) kann nicht geöffnet werden: "basetsd.h": No such file or directory
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit status 2
```

Any suggestions how to fix this?

**Comment by fkromer** [comment]
I created a PR with windows build instructions: https://github.com/welchbj/tt/pull/7 I don't know how to [consider the version file correctly yet](https://github.com/welchbj/tt/pull/7/files#r375246467). Probably someone can help out with this. However the build is possible and successful.

**Comment by welchbj** [comment]
Thanks for supporting this project! Sorry it took me a little while to get back to you. It looks like the build environment I had setup on AppVeyor wasn't compatible with Python 3.7, but you fixed that in #7.

It looks like the only issues in the CI builds for #7 are stemming from some changes in flake8 that mark some of the string escapes I used as linting errors. I'll clean those up this weekend and loot into the `_picosat.VERSION` issue.

Merging #7 shortly! Thanks again!
